pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -301,6 +308,36 @@ func (r *InferenceServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return reconcile.Result{}, err
 	}
 
+	if strings.EqualFold(forceStopRuntime, "true") {
+		// Delete the service
+		existingService := &corev1.Service{}
+		if err := r.Get(ctx, types.NamespacedName{Name: isvc.Name, Namespace: isvc.Namespace}, existingService); err != nil {
+			if apierr.IsNotFound(err) {
+				return ctrl.Result{}, nil
+			}
+			return ctrl.Result{}, err
+		} else {
+			r.Log.Info("Delete services")
+			if err := r.Delete(ctx, existingService); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		// Delete the virtualservice
```

Comment:
I’m not sure if it will work, as the VirtualService might be recreated by another reconciler. Have you had a chance to test it? I believe the VirtualService is managed by the ingress reconciler.

Reply:
yes tested and it was deleted and stayed deleted on my end

Reply:
Still, I am not sure it is a good idea to delete the object from here because the object should be managed by the right reconciller. The VS are managed by ingress reconcile, right? could we think to manage it from there?